src/cljc/athens/parser.cljc

```diff
@@ -38,13 +50,24 @@
    <url-link-url-part> = (backslash-escaped-paren | '(' url-link-url-part* ')') / any-char
    <backslash-escaped-paren> = <'\\\\'> ('(' | ')')
    
-   bold = <'**'> any-chars <'**'>
+   (* The following regular expression expresses this: (any character except '*') <- This repeated as many times as possible *)
+   <non-bold-chars> = #'[^\\*]*'
+   bold = <'**'> non-bold-chars <'**'>
    
-   (* It’s useful to extract this rule because its transform joins the individual characters everywhere it’s used. *)
-   (* However, I think in many cases a more specific rule can be used. So we will migrate away from uses of this rule. *)
-   any-chars = any-char+
+   (* -- It’s useful to extract this rule because its transform joins the individual characters everywhere it’s used. *)
+   (* -- However, I think in many cases a more specific rule can be used. So we will migrate away from uses of this rule. *)
    
+   (* Here are a list of 'stop characters' we implemented, to get the LL(1) performance. *)
```

Comment:
Can you elaborate a bit on the difference between comments that begin with `--`?